.eslintrc.json

```diff
@@ -34,6 +34,7 @@
     "no-throw-literal": "error",
     "no-void": "error",
     "@typescript-eslint/no-extra-semi": "off", // Prettier work
+    "no-return-await": "error",
```

Comment:
From ESLint docs ->
> You can avoid the extra microtask by not awaiting the return value, with the trade off of the function no longer being a part of the stack trace if an error is thrown asynchronously from the Promise being returned. This can make debugging more difficult.

Have we taken into account the "make debugging more difficult" tradeoff? I'm not sure how this changes the stacktrace, so just want to know if this was considered.